pygmt/clib/session.py

```diff
@@ -930,39 +931,43 @@ def put_vector(self, dataset, column, vector):
                 f"in column {column} of dataset."
             )
 
-    def put_strings(self, dataset, family, strings):
+    def put_strings(
+        self,
+        dataset: ctp.c_void_p,
+        family: Literal["GMT_IS_VECTOR", "GMT_IS_MATRIX"],
+        strings: StringArrayTypes,
+    ):
         """
-        Attach a numpy 1-D array of dtype str as a column on a GMT dataset.
+        Attach a 1-D numpy array of dtype str or pyarrow.StringArray as a column on a
+        GMT dataset.
 
-        Use this function to attach string type numpy array data to a GMT
-        dataset and pass it to GMT modules. Wraps ``GMT_Put_Strings``.
+        Use this function to attach string type array data to a GMT dataset and pass it
+        to GMT modules. Wraps ``GMT_Put_Strings``.
 
-        The dataset must be created by :meth:`pygmt.clib.Session.create_data`
-        first.
+        The dataset must be created by :meth:`pygmt.clib.Session.create_data` first.
 
         .. warning::
-            The numpy array must be C contiguous in memory. If it comes from a
-            column slice of a 2-D array, for example, you will have to make a
-            copy. Use :func:`numpy.ascontiguousarray` to make sure your vector
-            is contiguous (it won't copy if it already is).
+            The array must be C contiguous in memory. If it comes from a column slice of
+            a 2-D array, for example, you will have to make a copy. Use
+            :func:`numpy.ascontiguousarray` to make sure your vector is contiguous (it
+            won't copy if it already is).
 
         Parameters
         ----------
-        dataset : :class:`ctypes.c_void_p`
+        dataset
             The ctypes void pointer to a ``GMT_Dataset``. Create it with
             :meth:`pygmt.clib.Session.create_data`.
```

Comment:
When calling `create_data`, we use `family="GMT_IS_DATASET|GMT_VIA_VECTOR"`. The function actually creates a `GMT_VECTOR` container (a simplified version of the `GMT_DATASET` container) and returns the pointer.
```suggestion
        dataset
            The ctypes void pointer to a ``GMT_VECTOR``/``GMT_MATRIX`` container. 
            Create it with :meth:`pygmt.clib.Session.create_data`.
```